### ISSUE TITLE
Suggestion: Add game controller actions 

### ISSUE THREAD

**Issue by imkrut** [Martenfur/TabletFriend#28]
Basically be able to map a buton (or button combos) to a on-screen overlay button

**Comment by Martenfur** [comment]
y tho

**Comment by imkrut** [comment]
> y tho

In my particular case, I am streaming to my Switch with SkyNX, which maps the Xinput api (Xbox family) controllers to the Switch.
However, Switch's home and screen capture buttons are still mapped to their respective functions.

On PC I can press home and a number of buttons on my Xbox gamepad to create "new presses" on different software (for example on emulators, I can press Home + LS and I can load a state and Home + RS saves state).

With this in mind, I can create a "Save state" "Load State" "Fast forward" , etc button on the Switch touch screen (which I can easily access).

Retroarch already offers something like this, thanks to the "overlays", however, external emulators like PCSX2 or Dolphin, do not.

Another use comes to mind with button combos, for example, Marvel Vs. Capcom.  Say I could create a 3 button key press that allows me to do a super move or change characters with a single button press on the touch screen (which is right next to the gamepad anyway).

Those are some ideas that come to mind anyway.

**Comment by imkrut** [comment]
Makes sense. I wouldn't know where to start tho, completely code ignorant.

Thanks for the input tho.

**Comment by Martenfur** [comment]
Oh, then yeah, you're not gonna be able to do that. Faking keypresses requires a lot of black magic. Especially for controllers. What you can do is get some other software that maps gamepad keys to keyboard and use that together with Tabletfriend. Should work just fine.
